Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -274,22 +262,13 @@ extension BrowserViewController: URLBarDelegate {
     func urlBarDidPressReaderMode(urlBar: URLBarView) {
         if let tab = tabManager.selectedTab {
             if let readerMode = tab.getHelper(name: "ReaderMode") as? ReaderMode {
-                if readerMode.state == .Available {
-                    // Update the style of the reader mode to what was last configured
-                    if let dict = self.profile.prefs.dictionaryForKey(ReaderModeProfileKeyStyle) {
-                        if let style = ReaderModeStyle(dict: dict) {
-                            readerMode.style = style
-                        }
-                    }
-                    readerMode.enableReaderMode()
-
-//                    hideToolbars(animated: true, completion: { finished in
-//                        //self.showReaderModeBar(animated: false)
-//                    })
-                    self.showReaderModeBar(animated: true)
-                } else {
-                    readerMode.disableReaderMode()
-                    self.hideReaderModeBar(animated: true)
+                switch readerMode.state {
+                case .Available:
+                    enableReaderMode()
+                case .Active:
+                    disableReaderMode()
+                case .Unavailable:
+                    break
```

Comment:
This is something that should never happen, so I think we should be strict about it. Instead of an if statement, how about we assert this?